rust/common/redis/src/lib.rs

```diff
@@ -130,7 +196,21 @@ impl Client for RedisClient {
         v: String,
         seconds: u64,
     ) -> Result<bool, CustomRedisError> {
-        let bytes = serde_pickle::to_vec(&v, Default::default())?;
+        self.set_nx_ex_with_format(k, v, seconds, RedisValueFormat::Pickle)
+            .await
+    }
+
+    async fn set_nx_ex_with_format(
```

Comment:
Ha! Rust or Redis can be cryptic sometimes. I understand this to mean "Set if not exists with expiration with a format"

Reply:
Yeah exactly